### PR TITLE
fix: do some typing in commands/context.py

### PR DESCRIPTION
**Pull request by AnimateShadows**
## Summary

This PR just fixes some typing issues that I came across. (Specifically in `commands/context.py`)

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

File: discord/commands/context.py [hunk 1]
```diff
@@ -142,8 +148,8 @@ def guild_locale(self) -> Optional[str]:
         return self.interaction.guild_locale
 
     @cached_property
-    def me(self) -> Union[Member, User]:
-        return self.guild.me if self.guild is not None else self.bot.user
+    def me(self) -> Union[Member, ClientUser, None]:
```
Comment (Dorukyum) [inline discord/commands/context.py:151]: Wouldn't it be better to use Optional here?

File: discord/commands/context.py [hunk 2]
```diff
@@ -249,7 +253,7 @@ class AutocompleteContext:
 
     __slots__ = ("bot", "interaction", "command", "focused", "value", "options")
 
-    def __init__(self, bot: Bot, interaction: Interaction) -> None:
+    def __init__(self, bot: Bot, interaction: Interaction):
```
Comment (Dorukyum) [inline discord/commands/context.py:256]: Why is this removed?

File: discord/commands/context.py (hunk 1 above)
Comment (AnimateShadows) [reply discord/commands/context.py:151]: Optional is just an alias for Union[T, None] and unions get flattened anyway. I can change it if you want.

File: discord/commands/context.py (hunk 2 above)
Comment (AnimateShadows) [reply discord/commands/context.py:256]: Consistency

File: discord/commands/context.py (hunk 1 above)
Comment (Dorukyum) [reply discord/commands/context.py:151]: I said it for consistency with the other attributes, so yeah

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Comment by krittick** [comment]
Why did this revert the changes to `author` that were made in #464? This broke the typing that #464 fixed.

**Comment by krittick** [comment]
Specifically, this PR now causes this typing error which did not occur before the change:

image

**Comment by Lulalaby** [comment]
wtf

**Comment by AnimateShadows** [comment]
Why are we using `@cached_property` anyway
